### ISSUE TITLE
How to extract TSM features (8 frames as input)?

### ISSUE THREAD

**Issue by kakaguya** [assembly-101/assembly101-action-anticipation#1]
Hello, I noticed that you wrote "TempAgg requires per-frame features as input. TSM (8-frame input) has been used for extracting 2048-D per-frame features" in the Feature section. I would like to know in what form the 8 frames are input during the actual feature extraction process, as well as the more specific single-frame feature extraction strategy. Can you please give me an answer? Thanks!

**Comment by dibschat** [comment]
Hi Tokisaki,

Please consider referring to issues [#1](https://github.com/assembly-101/assembly101-action-recognition/issues/1) and [#6](https://github.com/assembly-101/assembly101-action-recognition/issues/6) of the [action recognition repo](https://github.com/assembly-101/assembly101-action-recognition).

TL;DR: We use the 2D backbone of a pretrained TSM (trained on video with shifting operations) for extracting the per-frame features. We repurpose the [RU-LSTM feature extraction script](https://github.com/fpv-iplab/rulstm/blob/master/FEATEXT/extract_example_rgb.py), by replacing TSN with TSM and using TSM image transforms.

Please let us know if you face any issues.

Best,
Dibyadip

**Comment by kakaguya** [comment]
Hi！
Thank you very much for your reply! I have learned the solution to the problem through your reply!
Best,
Tokisak
